integration_tests/src/main/python/parquet_testing_test.py

```diff
@@ -72,6 +73,43 @@
     _error_files["lz4_raw_compressed.parquet"] = "Exception"
     _error_files["lz4_raw_compressed_larger.parquet"] = "Exception"
 
+def hdfs_glob(path, pattern):
+    """
+    Finds hdfs files by checking the input path with glob pattern
+
+    :param path: hdfs path to check 
+    :type path: pathlib.Path 
+    :return: generator of matched files
+    """
+    path_str = path.as_posix()
+    full_pattern = path_str + '/' + pattern
+    cmd = ['hadoop', 'fs', '-ls', '-C', full_pattern]
```

Comment:
it should be possible to call FileSystem globStatus directly via PY4J without forking the JVM via `hadoop fs`

Reply:
Hi Gera, Thanks for the review! 

I tried following code to use PY4J to glob files by patterns and it works as well. 
But seems it's not so straightforward like `hadoop fs -ls` command and also we need to process the path string further. 

Considering the readability, I think probably we could keep the current implementation? 
```suggestion
    path_str = path.as_posix()
    full_pattern = path_str + '/' + pattern
    sc = get_spark_i_know_what_i_am_doing().sparkContext
    config = sc._jsc.hadoopConfiguration()
    fs_path = sc._jvm.org.apache.hadoop.fs.Path(full_pattern)
    fs = sc._jvm.org.apache.hadoop.fs.FileSystem.get(config)
    statuses = fs.globStatus(fs_path)

    for status in statuses:
        # status.getPath().toString() return string like "hdfs://hostname:8020/src/test/resources/parquet-testing/data/single_nan.parquet"
        # but pathlib.Path will remove the first "/" and convert it to "hdfs:/hostname:8020/src/test/resources/parquet-testing/data/single_nan.parquet" and then this path becomes illegal. 
        # so we need to process the path like this.
        p = f'hdfs:{status.getPath().toUri().getPath()}'
        yield Path(p)
```

Reply:
I find it readable enough. Not a must-fix but probably would save us trouble-shooting various OOMs and other resource issues down the road. We can file it as a potential improvement and mitigation.

Reply:
I will file it as a improvement. Thanks a lot!